lang/machine.md

```diff
@@ -45,12 +45,17 @@ We also define some helper functions that will be useful later.
 
 ```rust
 impl<M: Memory> Machine<M> {
-    fn cur_frame(&self) -> &StackFrame<M> {
+    fn cur_frame(&self) -> StackFrame<M> {
```

Comment:
This is again necessary, as we can't return refs into `GC_STATE`, and also it's not too costly as `StackFrame` contains just a bunch of usizes.